gitops/snapshot.go

```diff
@@ -122,6 +128,26 @@ func SetSnapshotIntegrationStatusAsInvalid(snapshot *applicationapiv1alpha1.Snap
 	})
 }
 
+// SetSnapshotIntegrationStatusAsInProgress sets the HACBS integration status condition for the Snapshot to In Progress.
+func SetSnapshotIntegrationStatusAsInProgress(snapshot *applicationapiv1alpha1.Snapshot, message string) {
+	meta.SetStatusCondition(&snapshot.Status.Conditions, metav1.Condition{
+		Type:    HACBSIntegrationStatusCondition,
+		Status:  metav1.ConditionFalse,
+		Reason:  HACBSIntegrationStatusInProgress,
+		Message: message,
+	})
+}
+
+// SetSnapshotIntegrationStatusAsFinished sets the HACBS integration status condition for the Snapshot to Finished.
+func SetSnapshotIntegrationStatusAsFinished(snapshot *applicationapiv1alpha1.Snapshot, message string) {
+	meta.SetStatusCondition(&snapshot.Status.Conditions, metav1.Condition{
+		Type:    HACBSIntegrationStatusCondition,
+		Status:  metav1.ConditionFalse,
```

Comment:
I think we might want to set the status to `metav1.ConditionTrue` here since this is the expected end goal of this condition.

Reply:
Thanks. So `metav1.ConditionTrue` means it is expected while `metav1.ConditionFalse` means it is unexpected? I searched the `meta` but I didn't find when we should use `True` or `False`. Thanks!

Reply:
@dirgim I found these help comments in `vendor` dir, is it?
```
// These are valid condition statuses. "ConditionTrue" means a resource is in the condition.
// "ConditionFalse" means a resource is not in the condition. "ConditionUnknown" means kubernetes
// can't decide if a resource is in the condition or not. In the future, we could add other
// intermediate conditions, e.g. ConditionDegraded.
```

Reply:
Yes, I think we're on the right track - setting the `ConditionTrue` when the integration service successfully processes the snapshot seem to be in line with the guidance.

Reply:
Updated. Thanks.